augur/errors.py

```diff
@@ -1,2 +1,16 @@
 class AugurError(Exception):
     pass
+
+
+# TODO: Move this to something like dates/errors.py
+class InvalidDate(Exception):
+    """Custom exception class to handle dates in unsupported formats."""
+
+    def __init__(self, date, message):
+        """Initialize the error with the culprit date string and a meaningful message."""
+        self.date = date
+        self.message = message
```

Comment:
I expected to see `super().__init__(message)` here, but that might just be my old-school memory of Python classes. The `Exception` class doesn't store the message in its own attribute but in a tuple `args` attribute, so if you used this approach, your `__str__` method would need to reference `self.args[0]` instead of `self.message`.

Reply:
The difference in readability between `self.args[0]` vs. `self.message` is why I chose setting `self.message` over `super().__init__(message)`.

The class needs to inherit from an existing Exception class, but there's no need to call `super().__init__()`. It could still be there, but it wouldn't be used, so might as well leave it out.